src/parser/mod.rs

```diff
@@ -10533,8 +10533,17 @@ impl<'a> Parser<'a> {
     }
 
     pub fn parse_set(&mut self) -> Result<Statement, ParserError> {
-        let modifier =
-            self.parse_one_of_keywords(&[Keyword::SESSION, Keyword::LOCAL, Keyword::HIVEVAR]);
+        let modifier_keywords = if self.dialect.supports_global_variable_modifier() {
```

Comment:
I'm thinking having the lone `global` keyword as a dialect flag might not scale API wise, to make it dialect specific maybe it might make sense to have delegate to the dialect? like with [parse_column_option](https://github.com/apache/datafusion-sqlparser-rs/blob/c0b5b20d11cee97c9541e6671a9833e1c2f926e7/src/dialect/mod.rs#L622) as an example except with a default implementation that others like hive and mysql can override. 
e.g. 
```rust
let scope = self.dialect.parse_set_variable_scope(self)?;
```